lib/foreman_tasks.rb

```diff
@@ -20,16 +20,27 @@ def self.trigger(action, *args, &block)
   end
 
   def self.trigger_task(async, action, *args, &block)
-    Match! async, true, false
-
-    match trigger(action, *args, &block),
-          (on ::Dynflow::World::PlaningFailed.call(error: ~any) do |error|
-            raise error
-          end),
-          (on ::Dynflow::World::Triggered.call(execution_plan_id: ~any, future: ~any) do |id, finished|
-            finished.wait if async == false
-            ForemanTasks::Task::DynflowTask.where(:external_id => id).first!
-          end)
+    rails_safe_trigger_task do
+      Match! async, true, false
+      match trigger(action, *args, &block),
+            (on ::Dynflow::World::PlaningFailed.call(error: ~any) do |error|
+              raise error
+            end),
+            (on ::Dynflow::World::Triggered.call(execution_plan_id: ~any, future: ~any) do |id, finished|
+              finished.wait if async == false
+              ForemanTasks::Task::DynflowTask.where(:external_id => id).first!
+            end)
+    end
+  end
+
+  def self.rails_safe_trigger_task
+    if SETTINGS[:rails] == '4.2'
```

Comment:
Checking on `Rails::VERSION::MAJOR` would be better, as it would not depend on the rails setting being present.